### PR TITLE
PYIC-2509: Switch to using the new config references in core-back

### PR DESCRIPTION
**Pull request by shivanshuit914**
### Issue tracking

- [PYIC-2509](https://govukverify.atlassian.net/browse/PYIC-2509)

### Environment variables or secrets

- [x] No environment variables or secrets were added or changed

[PYIC-2509]: https://govukverify.atlassian.net/browse/PYIC-2509?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: lambdas/select-cri/src/main/java/uk/gov/di/ipv/core/selectcri/SelectCriHandler.java [hunk 1]
```diff
@@ -320,22 +322,25 @@ private Optional<JourneyResponse> getCriResponse(
     }
 
     private Optional<JourneyResponse> getMultipleDocCheckPage() {
-        if (drivingLicenceCriId.startsWith("stub")) {
+        CredentialIssuerConfig drivingLicenceConfig =
+                configService.getCredentialIssuerActiveConnectionConfig(drivingLicenceCriId);
+        if (drivingLicenceConfig.getActiveConnection() == "stub") {
```
Comment (AmritSidhu) [inline lambdas/select-cri/src/main/java/uk/gov/di/ipv/core/selectcri/SelectCriHandler.java]: This wont work as activeConnection is a layer above, may need to use the original getCredentialIssuer method for this one

File: lambdas/validate-oauth-callback/src/main/java/uk/gov/di/ipv/core/validateoauthcallback/ValidateOAuthCallbackHandler.java [hunk 2]
```diff
@@ -199,9 +199,9 @@ private Map<String, Object> sendOauthErrorJourneyResponse(
 
         if (OAuth2Error.ACCESS_DENIED_CODE.equals(error)) {
             CredentialIssuerConfig passportCriConfig =
-                    configService.getCredentialIssuer(passportCriId);
+                    configService.getCredentialIssuerActiveConnectionConfig(passportCriId);
             CredentialIssuerConfig ukDrivingLicenseCriConfig =
-                    configService.getCredentialIssuer(drivingLicenceCriId);
+                    configService.getCredentialIssuerActiveConnectionConfig(drivingLicenceCriId);
             if (passportCriConfig.getEnabled() && ukDrivingLicenseCriConfig.getEnabled()) {
```
Comment (AmritSidhu) [inline lambdas/validate-oauth-callback/src/main/java/uk/gov/di/ipv/core/validateoauthcallback/ValidateOAuthCallbackHandler.java]: This needs to be looked at aswell, i dont beleive getEnabled would work, so again may have to switch to using the getCredentialIssuer method?

File: lib/src/main/java/uk/gov/di/ipv/core/library/config/ConfigurationVariable.java [hunk 3]
```diff
@@ -12,6 +12,7 @@ public enum ConfigurationVariable {
     FRAUD_CRI_ID("/%s/core/self/journey/fraudCriId"),
     KBV_CRI_ID("/%s/core/self/journey/kbvCriId"),
     DCMAW_CRI_ID("/%s/core/self/journey/dcmawCriId"),
+    DCMAW_CRI_ID_ID("/%s/core/self/journey/dcmawCriId"),
```
Comment (AmritSidhu) [inline lib/src/main/java/uk/gov/di/ipv/core/library/config/ConfigurationVariable.java]: to be removed

File: lib/src/main/java/uk/gov/di/ipv/core/library/service/ConfigService.java [hunk 4]
```diff
@@ -106,6 +106,8 @@ public String getSsmParameter(ConfigurationVariable configurationVariable) {
                         configurationVariable.getValue(), getEnvironmentVariable(ENVIRONMENT)));
     }
 
+    // do we add another getSsm to pass in the active connection
```
Comment (AmritSidhu) [inline lib/src/main/java/uk/gov/di/ipv/core/library/service/ConfigService.java]: to be removed

File: lambdas/build-cri-oauth-request/src/main/java/uk/gov/di/ipv/core/buildcrioauthrequest/BuildCriOauthRequestHandler.java [hunk 5]
```diff
@@ -150,7 +150,8 @@ public APIGatewayProxyResponseEvent handleRequest(
                             userId,
                             oauthState,
                             govukSigninJourneyId,
-                            currentVcStatuses);
+                            currentVcStatuses,
+                            criId);
```
Comment (AmritSidhu) [inline lambdas/build-cri-oauth-request/src/main/java/uk/gov/di/ipv/core/buildcrioauthrequest/BuildCriOauthRequestHandler.java:154]: Why has criId been added here?

File: lambdas/build-cri-oauth-request/src/main/java/uk/gov/di/ipv/core/buildcrioauthrequest/BuildCriOauthRequestHandler.java [hunk 6]
```diff
@@ -213,10 +214,11 @@ private JWEObject signEncryptJar(
             String userId,
             String oauthState,
             String govukSigninJourneyId,
-            List<VcStatusDto> currentVcStatuses)
+            List<VcStatusDto> currentVcStatuses,
+            String criId)
```
Comment (AmritSidhu) [inline lambdas/build-cri-oauth-request/src/main/java/uk/gov/di/ipv/core/buildcrioauthrequest/BuildCriOauthRequestHandler.java:218]: Why has criId been added here?

File: lib/src/main/java/uk/gov/di/ipv/core/library/service/ConfigService.java [hunk 7]
```diff
@@ -140,14 +142,21 @@ public String getSigningKeyId() {
...
                         String.format(
-                                "%s/%s",
+                                "%s/%s/connections/%s",
                                 getEnvironmentVariable(CREDENTIAL_ISSUERS_CONFIG_PARAM_PREFIX),
-                                credentialIssuerId));
-        return new ObjectMapper().convertValue(result, CredentialIssuerConfig.class);
+                                credentialIssuerId,
+                                getActiveConnection(credentialIssuerId)));
+
+        CredentialIssuerConfig credentialIssuerConfig =
```
Comment (AmritSidhu) [inline lib/src/main/java/uk/gov/di/ipv/core/library/service/ConfigService.java:153]: Didnt we want to give rid of this and make changes in the code to reference the Id other ways?

File: lib/src/test/java/uk/gov/di/ipv/core/library/service/ConfigServiceTest.java [hunk 8]
```diff
@@ -34,10 +34,7 @@
 import static com.github.tomakehurst.wiremock.client.WireMock.ok;
 import static com.github.tomakehurst.wiremock.client.WireMock.post;
 import static com.github.tomakehurst.wiremock.client.WireMock.stubFor;
-import static org.junit.jupiter.api.Assertions.assertEquals;
-import static org.junit.jupiter.api.Assertions.assertNull;
-import static org.junit.jupiter.api.Assertions.assertThrows;
-import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.*;
```
Comment (AmritSidhu) [inline lib/src/test/java/uk/gov/di/ipv/core/library/service/ConfigServiceTest.java]: We dont use the *, we typically define the imports we use

File: lib/src/main/java/uk/gov/di/ipv/core/library/service/ConfigService.java (hunk 7 above)
Comment (shivanshuit914) [reply lib/src/main/java/uk/gov/di/ipv/core/library/service/ConfigService.java:153]: Yes, I looked into code and its been used many places, thinking to separate PR for it.

File: lambdas/build-cri-oauth-request/src/main/java/uk/gov/di/ipv/core/buildcrioauthrequest/BuildCriOauthRequestHandler.java (hunk 6 above)
Comment (shivanshuit914) [reply lambdas/build-cri-oauth-request/src/main/java/uk/gov/di/ipv/core/buildcrioauthrequest/BuildCriOauthRequestHandler.java:218]: We need to pass criId to get getAllowedSharedAttributes. just passing from top level function so we can get it from config service.

File: lambdas/build-cri-oauth-request/src/main/java/uk/gov/di/ipv/core/buildcrioauthrequest/BuildCriOauthRequestHandler.java (hunk 5 above)
Comment (shivanshuit914) [reply lambdas/build-cri-oauth-request/src/main/java/uk/gov/di/ipv/core/buildcrioauthrequest/BuildCriOauthRequestHandler.java:154]: Just passing from higher level function so we can get `getAllowedSharedAttributes`

**Review by AmritSidhu — APPROVED** [review]

File: lib/src/main/java/uk/gov/di/ipv/core/library/dto/CredentialIssuerConfig.java [hunk 9]
```diff
@@ -25,27 +24,23 @@ public class CredentialIssuerConfig {
     private String jarEncryptionPublicJwk;
     private String audienceForClients;
     private URI ipvCoreRedirectUrl;
-    private String allowedSharedAttributes;
```
Comment (TalatAydinGDS) [inline lib/src/main/java/uk/gov/di/ipv/core/library/dto/CredentialIssuerConfig.java:28]: We added this property in https://govukverify.atlassian.net/browse/PYIC-2566. We are planning to read all configuration from this path credentialIssuers/criId/connections. I have some questions about this changes. Will we keep to all CredentialIssuerConfig params in this path credentialIssuers/criId or not? We have duplicate keys and confusing in the future. Can we discuss it?

File: lib/src/main/java/uk/gov/di/ipv/core/library/service/ConfigService.java [hunk 10]
```diff
@@ -182,6 +189,44 @@ public String getCriPrivateApiKey(String criId) {
...
 
+    public String getActiveConnection(String credentialIssuerId) {
+        return getSsmParameter(
+                String.format(
+                        "%s/%s/activeConnection",
+                        getEnvironmentVariable(CREDENTIAL_ISSUERS_CONFIG_PARAM_PREFIX),
+                        credentialIssuerId));
+    }
+
+    public boolean isUnavailable(String credentialIssuerId) {
```
Comment (TalatAydinGDS) [inline lib/src/main/java/uk/gov/di/ipv/core/library/service/ConfigService.java:200]: I think, we can add generic method for all 3 methods and get this value unavailable, allowedsharedAttributes, enabled.. from second method param.

File: lib/src/main/java/uk/gov/di/ipv/core/library/service/ConfigService.java [hunk 11]
```diff
@@ -140,14 +140,21 @@ public String getSigningKeyId() {
...
     }
 
-    public CredentialIssuerConfig getCredentialIssuer(String credentialIssuerId) {
+    public CredentialIssuerConfig getCredentialIssuerActiveConnectionConfig(
+            String credentialIssuerId) {
         Map<String, String> result =
                 getSsmParameters(
                         String.format(
-                                "%s/%s",
+                                "%s/%s/connections/%s",
```
Comment (TalatAydinGDS) [inline lib/src/main/java/uk/gov/di/ipv/core/library/service/ConfigService.java:148]: I thought, we can take connections static path from different params. So we can manage configuration changes with params and without code changes.

File: lib/src/main/java/uk/gov/di/ipv/core/library/dto/CredentialIssuerConfig.java (hunk 9 above)
Comment (TalatAydinGDS) [reply lib/src/main/java/uk/gov/di/ipv/core/library/dto/CredentialIssuerConfig.java:28]: Amrit and Shivanshu have a plan that keep some properties at the higher level not within connections. Next plan remove some properties with different PR at high level.

**Pull request merged**